backend/src/ApiRequest/DataApi.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Database.ApiRequest.Dto;
+using Database.Services;
+using GraphQL;
+using Microsoft.AspNetCore.Http;
+
+namespace Database.ApiRequest;
+
+/// <summary>
+/// Class to request XData from Database API.
+/// </summary>
+public class DataApi
+{
+    private static readonly string[] _calorimetricDataFileNames =
+    {
+        "CalorimetricData.graphql"
+    };
+
+    private static readonly string[] _geometricDataFileNames =
+    {
+        "GeometricData.graphql"
+    };
+
+    private static readonly string[] _hygrothermalDataFileNames =
+    {
+        "HygrothermalData.graphql"
+    };
+
+    private static readonly string[] _photovoltaicDataFileNames =
+    {
+        "PhotovoltaicData.graphql"
+    };
+
+    private static readonly string[] _opticalDataFileNames =
+    {
+        "OpticalData.graphql"
+    };
+
+    private sealed record CalorimetricDataResponse(CalorimetricDataDto CalorimetricData);
+    private sealed record GeometricDataResponse(GeometricDataDto GeometricData);
+    private sealed record HygrothermalDataResponse(HygrothermalDataDto HygrothermalData);
+    private sealed record PhotovoltaicDataResponse(PhotovoltaicDataDto PhotovoltaicData);
+    private sealed record OpticalDataResponse(OpticalDataDto OpticalData);
+
+    /// <summary>
+    /// Create query to request calorimetric data and get response.
+    /// </summary>
+    /// <param name="dataId">              Id of data to request. </param>
+    /// <param name="appSettings">         <see cref="AppSettings"/> </param>
+    /// <param name="apiRequestService">   <see cref="IApiRequestService"/> </param>
+    /// <param name="httpClientFactory">   <see cref="IHttpClientFactory"/> </param>
+    /// <param name="httpContextAccessor"> <see cref="IHttpContextAccessor"/> </param>
+    /// <param name="cancellationToken">   <see cref="CancellationToken"/> </param>
+    /// <returns> Query and response for calorimetric data. </returns>
+    public static async Task<(string? Query, string? Response)> CreateQueryAndGetResponseCalorimetricData(
```

Comment:
Remove the `?` for both strings here. And below store the result of `apiRequestService.ConstructGraphQlQuery` in the variable `query` and return that query instead of `request.Query`. Do the same for the methods below that follow the very same structure. The five almost identical methods can probably be simplified by just calling a generic version with some parameters.